apps/meteor/ee/app/livechat-enterprise/server/api/lib/priorities.ts

```
@@ -37,11 +37,25 @@ export async function findPriority({
 	};
 }
 
-export async function updatePriority(_id: string, data: Pick<ILivechatPriority, 'name'>): Promise<ILivechatPriority> {
+export async function updatePriority(
+	_id: string,
+	data: Pick<ILivechatPriority & { reset: boolean }, 'name' | 'reset'>,
+): Promise<ILivechatPriority | null> {
 	const query = {
 		_id,
 	};
+	const update: Pick<UpdateFilter<ILivechatPriority>, '$set' | '$unset'> = {
+		...((data.reset && {
+			$set: { dirty: false },
+			$unset: { name: 1 },
+		}) || {
+			$set: { name: data.name, dirty: true },
+		}),
+	};
+
+	const created = await LivechatPriority.findOneAndUpdate(query, update, {
+		returnDocument: 'after',
+	});
 
-	const created = await LivechatPriority.findOneAndUpdate(query, { $set: { ...data, dirty: true } }, { returnDocument: 'after' });
-	return created.value as ILivechatPriority;
+	return created.value;
```

You can put an `if (!created) throw Error('invalid-priority')` or smth :eyes: